ledger/sandbox/src/main/scala/com/digitalasset/platform/index/JdbcIndexer.scala

```diff
@@ -173,13 +175,12 @@ class PostgresIndexer private (
       case PartyAddedToParticipant(party, displayName, _, _) =>
         ledgerDao.storeParty(party, Some(displayName)).map(_ => ())(DEC)
 
-      case PublicPackageUploaded(archive, sourceDescription, _, _) =>
+      case PublicPackageUploaded(archive, sourceDescription, _, recordTime) =>
         val uploadId = UUID.randomUUID().toString
-        val uploadInstant = Instant.now()
         val packages: List[(DamlLf.Archive, v2.PackageDetails)] = List(
           archive -> v2.PackageDetails(
             size = archive.getPayload.size.toLong,
-            knownSince = uploadInstant,
+            knownSince = recordTime.toInstant,
```

Comment:
I'm also not 100% sure it's correct. I remember talking to @gerolf-da about this and that we should use the actual upload time rather than the time provided by the committer.

Reply:
I will open a separate issue for RecordTime. Multi-ledgers such as Canton and Corda are allowed to "opt out" of recordTime (which is being made optional for multi-ledgers). The mechansim to opt-out is to send Epoch-time instead (which Canton does by setting an recordTime fields to Timestamp.Epoch on all event types which expose recordTime.

Reply:
Tracked as #2635 and removing change from this PR.